conf/script/v2/pre_disruptive_upgrade

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+set -eu -o pipefail
+
+log_file=/var/log/cortx-ha.log
+BACKUP_SOURCE_DIR=/etc/cortx/ha
+BACKUP_DEST_DIR=/opt/seagate/cortx/ha_conf_backup
+consul=/usr/bin/consul
+HARE_DIR=/var/lib/hare
+pcs=/usr/sbin/pcs
+crm_resource=/usr/sbin/crm_resource
+crm_standby=/usr/sbin/crm_standby
+
+exec &>> $log_file
+exec &> >(stdbuf -oL gawk '{ print strftime("%Y-%m-%d %H:%M:%S"), $0 }')
+
+log() {
+    logger --stderr --tag pre_disruptive_upgrade "$*"
+}
+
+log "****************************************************************"
+log "${0##*/}"
+log "****************************************************************"
+
+PROG=${0##*/}
+
+usage() {
+    cat <<EOF
+Usage: $PROG
+
+* The script expects existing Pacemaker cluster to be running.
+
+Optional parameters:
+    <-cb>         Take consul backup
+EOF
+}
+
+backup_consul(){
+
+    echo "Taking the backup of consul"
+    log "${FUNCNAME[0]}: Exporting the Consul KV"
+
+    consul_kv_dump=$HARE_DIR/consul-kv-dump.json
+    if [[ -f $consul_kv_dump ]]; then
+        # Facing one codacy issue with below line. Hence, compressing the file in a
+        # different way
+        #cat $consul_kv_dump | xz > ${consul_kv_dump%.json}-$(date '+%Y%m%d%H%M%S').json.xz
+
+        tar -czf ${consul_kv_dump%.json}-$(date '+%Y%m%d%H%M%S').json.tar.gz $consul_kv_dump
```

Comment:
This will keep on accumulating files. We should have just one copy. Older can be deleted.

Reply:
Done.